### PR TITLE
bluetooth: controller: radio: Use HAL API in radio_nrf5_ppi

### PR DESCRIPTION
**Pull request by auroraslb**
Rewrite some functions in radio_nrf5_ppi to use available HAL APIs for subscribe and publish. This will make it possible to test the controller in a simulator. Also add guards on TXPOWER to ensure positive values only are used if they are defined.

### CODE REVIEW

**Review by cvinayak — CHANGES_REQUESTED** [review]
Good work.

Some change requests:
1. Make a separate PR for the TX power commit, so it can be approved independently.
2. Add a new commit, which moves the nRF53 DPPI to radio_nrf5_dppi.h (related to https://github.com/zephyrproject-rtos/zephyr/issues/26118)

**Comment by auroraslb** [comment]
> Good work.
> 
> Some change requests:
> 
>     1. Make a separate PR for the TX power commit, so it can be approved independently.
> 
>     2. Add a new commit, which moves the nRF53 DPPI to radio_nrf5_dppi.h (related to #26118)

Hi, I've created a separate PR for the TX power changes (#30880).
I will look more into a new commit, which moves the nRF53 DPPI to radio_nrf5_dppi.h.

**Comment by cvinayak** [comment]
@auroraslb FYI, I have cherry-picked your changes with some changes here #30950, do review and take the changes as necessary/appropriate into this PR.

**Comment by carlescufi** [comment]
@auroraslb will you follow-up on this PR?

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h [hunk 1]
```diff
@@ -40,5 +40,10 @@
 #include "radio_nrf5340.h"
 #endif
 
+#if defined(PPI_PRESENT)
 #include "radio_nrf5_ppi.h"
+#endif
+#if defined(DPPI_PRESENT)
+#include "radio_nrf5_dppi.h"
+#endif
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h:48]:
```suggestion
#elif defined(DPPI_PRESENT)
#include "radio_nrf5_dppi.h"
#else
#error "PPI or DPPI abstractions missing."
#endif
```

**Review by cvinayak — CHANGES_REQUESTED** [review]
I see that my original change requests have been addressed. Makes sense that this PR gets merged prior to #31336. I can rebase #31336 thereafter.

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h [hunk 2]
```diff
@@ -40,5 +40,14 @@
 #include "radio_nrf5340.h"
 #endif
 
+#include <nrf_peripherals.h>
+
+#if defined(PPI_PRESENT)
 #include "radio_nrf5_ppi.h"
+#endif
+#if defined(DPPI_PRESENT)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h:48]:
```suggestion
#elif defined(DPPI_PRESENT)
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]

**Comment by cvinayak** [comment]
@rugeGerritsen any progress here? This is blocking #31336 and  #30708

**Comment by rugeGerritsen** [comment]
@cvinayak , all work in this PR is done. I see you have approved it. It is ready to be merged.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**